File: wlr/SPX.Collision/CoordVerifier.cs
using System.Linq;
using OGUR.Management;
using OGUR.Sprites;
using OGUR.GameObjects;

namespace OGUR.Collision
{
    public static class CoordVerifier
    {
        public static bool IsValid(Point2 position)
        {
            return (position.PosX >= 0 && position.PosY >= 0 && position.PosX < XnaManager.WindowWidth && position.PosY < XnaManager.WindowHeight);
        }

        public static bool IsBlocked(Point2 target)
        {
            return GameplayObjectManager.IsLocationBlocked(target);
        }   

        public static bool Contains(Point2 target, int type)
        {
            return GameplayObjectManager.AnyContains(target, type);
        }
    }
}

File: aigilas/src/sps/entities/CoordVerifier.java
package sps.entities;

import sps.bridge.EntityType;
import sps.core.Point2;
import sps.core.SpsConfig;
import sps.graphics.Renderer;

public class CoordVerifier {
    public static boolean isValid(Point2 position) {
        return (position.PosX >= 0 && position.PosY >= 0 && position.PosX < Renderer.get().VirtualWidth && position.PosY < Renderer.get().VirtualHeight && position.GridX < SpsConfig.get().tileMapWidth && position.GridY < SpsConfig.get().tileMapHeight);
    }

    public static boolean isBlocked(Point2 target)

    {
        return EntityManager.get().isLocationBlocked(target);
    }

    public static boolean contains(Point2 target, EntityType type)

    {
        return EntityManager.get().anyAt(target, type);
    }
}
